src/main/java/frc/robot/Limelight.java

```diff
@@ -22,4 +62,202 @@ public void togglePipeline() {
+     * @return A rotation2D made from the robot pitch and yaw values
+     */
+    public Rotation2d createRotation2D() {
+
+        /**
+         * Gets the pitch value from the robotPositionValues array & converts it to radians
+         */
+        double robotRotationPitch = getRobotPosition3D()[3].doubleValue();
+        double robotRotationPitchRadians = Math.toRadians(robotRotationPitch);
+
+        /**
+         * Gets the yaw value from the robotPositionValues array & converts it to radians
+         */
+        double robotRotationYaw = getRobotPosition3D()[4].doubleValue();
+        double robotRotationYawRadians = Math.toRadians(robotRotationYaw);
+
+        /**
+         * Uses the pitch & yaw value to construct a rotation2d, then returns it
+         */
+        return new Rotation2d(robotRotationPitchRadians, robotRotationYawRadians);
+
+    }
+
+    /**
+     * Uses values from networkTables to construct and return a translation2D
+     * @return a translation2D made from the robot x and robot y values
+     */
+    public Translation2d createTranslation2D() {
+        
+        /**
+         * Gets the x & y values from the robotPositionValues array
+         */
+        double robotPositionX = getRobotPosition3D()[0].doubleValue();
+        double robotPositionY = getRobotPosition3D()[1].doubleValue();
+
+        /**
+         * Uses the x & y values to construct a translation2d, then returns it
+         */
+        return new Translation2d(robotPositionX, robotPositionY);
+
+    }
+    /**
+     *  Uses an existing translation2d and rotation2d to make a transform2d
+     * @param constructorTranslation2d
+     * @param constructorRotation2d
+     * @return A transform2d (rotates and drives at the same time)
+     */
+    public Transform2d createTransform2D(Translation2d constructorTranslation2d, Rotation2d constructorRotation2d) {
+        /**
+         * Uses a translation2d & a rotation2d parameter to construct a transform2d, then returns it
+         */
+        return new Transform2d(constructorTranslation2d, constructorRotation2d);
+
+    }
+
+
+
+     /**
+      * returns a boolean value that lets us know if the limelight has any targets
+      * @return If the limelight has any targets
+      */
+    public boolean hasTargets() {
+        return table.getEntry("tv").getBoolean(false);
+    }
+
+
+     /**
+      *checks if the limelight is in Apriltag, and if it has a target,  returns the ID of the Apriltag. Otherwise, it returns null.
+      * @return The ID of the currently targeted Apriltag
+      */
+    public Double getTargetID(){
+        if(getPipeline() == 1)
+        {
+            if(hasTargets()){
+            return table.getEntry("tid").getDouble(0.0);
+            }
+         }
+
+        return null;
+
+    }
+
+    /**
+     * Checks what target we are looking at, calculates the distance, and returns it
+     * @return The Horizontal Distance from the current vision target
+     */
```

Comment:
elaborate on horizontal distance; there are multiple possible interpretations of this. this could mean horizontal distance along the floor or perpendicular to the target

Reply:
This method has been renamed to forwardDistanceToTarget() to make it more clear what it does